Sources/Confidence/EventStorage.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+internal protocol EventStorage {
+    func rollover() throws
+    func writeEvent(event: Event) throws
+    func batchReadyFiles() throws -> [URL]
+    func eventsFrom(fileURL: URL) throws -> [Event]
+}
+
+internal class EventStorageImpl: EventStorage {
+    let DIRECTORY = "events"
+    let EVENT_WRITE_DELIMITER = ",\n"
+    let READY_TO_SEND_EXTENSION = ".ready"
+    let encoder = JSONEncoder()
+    let decoder = JSONDecoder()
+    var folderURL: URL = URL(string: "")!
+    var fileURL: URL = URL(string: "")!
+    var currentBatch: [Event] = []
+
+    init() throws {
+        folderURL = try getFolderURL()
+        try createEventsDirectory(url: folderURL)
+        fileURL = folderURL.appendingPathComponent("events-\(Date().currentTime)")
+    }
+
+    func rollover() throws {
+        let newPath = URL(string: "\(fileURL)" + "\(Date().currentTime)")!
```

Comment:
Given that `fileURL` already contains a date, I think the newPath will have two dates after each other?